crates/sparrow-main/src/serve/file_service.rs

```diff
@@ -55,63 +54,53 @@ impl FileService for FileServiceImpl {
 }
 
 async fn get_metadata(
-    s3: S3Helper,
+    object_store_registry: Arc<ObjectStoreRegistry>,
     request: tonic::Request<GetMetadataRequest>,
 ) -> anyhow::Result<tonic::Response<GetMetadataResponse>> {
     let request = request.into_inner();
 
     let file_metadatas: Vec<_> = request
         .file_paths
         .iter()
-        .map(|source| get_source_metadata(&s3, source))
+        .map(|source| get_source_metadata(Some(object_store_registry.as_ref()), source))
```

Comment:
Any particular reason the registry is passed as optional? Feels like since we're creating it on startup, we know it'll exist.

Reply:
I originally made it optional since it made the tests not require the creation of it/passing but making it required simplifies the logic so I made those changes.